### DISCUSSION TITLE
Feature request: export types `Getter`, `Setter` and `Write`

### DISCUSSION THREAD

**Discussion by rodrigocfd** [pmndrs/jotai#2248] [Ideas] [answered]
In order to simplify the creation of [write-only atoms](https://github.com/pmndrs/jotai#write-only-derived-atoms), suppressing the initial `null` value, I wrote the following function:

```ts
export function atomAction<A extends unknown[], R>(write: Write<A, R>) {
	return atom(null, write);
}
```

This allows me to write stuff like this:

```ts
const namesAtom = atom([] as string[]);

const addNameAtom = atomAction((get, set, newName: string) => {
	set(namesAtom, [...get(namesAtom),  newName]);
});
```

However... the problem is that the types `Getter`, `Setter` and `Write` are not exported by the library, so I had to manually copy and redeclare them. This is my current implementation:

```ts
type Getter = <Value>(atom: Atom<Value>) => Value;
type Setter = <Value, Args extends unknown[], Result>(atom: WritableAtom<Value, Args, Result>, ...args: Args) => Result;
type Write<Args extends unknown[], Result> = (get: Getter, set: Setter, ...args: Args) => Result;

export function atomAction<A extends unknown[], R>(write: Write<A, R>) {
	return atom(null, write);
}
```

So, to make it easier to write such particular utility functions – I also wrote others for my particular needs –, it would be a lot nicer if `Getter`, `Setter` and `Write` were actually exported by the library.

**Comment by dai-shi** [comment, marked as answer]
Getter and Setter are exposed. Write isn't. Please define it on your end for the moment.

https://tsplay.dev/wXyBON
<img width="495" alt="image" src="https://github.com/pmndrs/jotai/assets/490574/9e624d06-b805-4634-b653-e87e6e3100c8">

**Comment by adaptive-shield-matrix** [comment]
I thought about doing something similar. 
Great work, Thanks!
